[chore]: v16 updated chanelog (#5418)
osmosis-labs/osmosis: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

> ## v16.0.0

Reviewer: Could you please also review our v16 upgrade handler and make sure that we specify all parameter updates in v16 here?

Reviewer: Please ping me on Slack once complete

Author: updated!

This release containts the following new modules and updates:

> - Concentrated Liquidity module (x/concentrated-liquidity). Concentrated liquidity is a novel Automated Market Maker (AMM) design introduced by Uniswap that allows for more efficient use of capital. The improvement is

Reviewer 2: ```suggestion
- Concentrated Liquidity module (x/concentrated-liquidity). Concentrated liquidity is a gamechanging new pool model that will bring unprecedented capital efficiency to Osmosis. The initial version of our design is heavily influenced by Uniswap V3, although our implementation has a number of important differences.
```

> achieved by providing liquidity in specific price ranges chosen by the user.

Reviewer 2: ```suggestion
```

> - implement Cosmwasmpool module (x/cosmwasmpool). The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts. 

Reviewer 2: ```suggestion
- Implement Cosmwasmpool module (x/cosmwasmpool). The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts. 
```

- ProtoRev changes 
  - Integrate with Concentrated Liquidity Pools. 
  - Changes dev account payment from once per week to after every trade
- Dependency Updates 

>   - bump wasmvm to v1.2 and add cosmwasm_1_2 capability

Reviewer: Could you please add that validators will have to update the wasm cache during the upgrade due to this:

https://osmosis-network.slack.com/archives/C027ELA290B/p1685870833358839?thread_ts=1685716114.561639&cid=C027ELA290B

  - update async ICQ version 

>   - apply Cosmwasm Cherry security patch 

Reviewer: Isn't cosmawsm cherry the same as "bump wasmvm to v1.2 and add cosmwasm_1_2 capability"?

  * [#4983](https://github.com/osmosis-labs/osmosis/pull/4983) implement gas consume on denom creation
  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Scale gas costs by denoms in gauge
  * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) feat: make epochs standalone
  * [#4320](https://github.com/osmosis-labs/osmosis/pull/4320) feat: crosschain registry contract 
  * [#4382](https://github.com/osmosis-labs/osmosis/pull/4382) Tokenfactory: Add Before send hooks
  * [#4801](https://github.com/osmosis-labs/osmosis/pull/4801) refactor: remove GetTotalShares, GetTotalLiquidity and GetExitFee from PoolI
  * [#4951](https://github.com/osmosis-labs/osmosis/pull/4951) feat: implement pool liquidity query in pool manager, deprecate the one in gamm

## Dependencies
  * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31 
  * [#5404](https://github.com/osmosis-labs/osmosis/pull/5404) Cosmwasm Cherry security patch 
  * [#5320](https://github.com/osmosis-labs/osmosis/pull/5320) minor: huckleberry ibc patch

### Misc Improvements
 

> * [#5356](https://github.com/osmosis-labs/osmosis/pull/5356) Fix wrong restHandler for ReplaceMigrationRecordsProposal

Reviewer: nit: I think indentation broke here

Author: oops fixed

  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
  * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0

>   * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) add amino proto annotations

Reviewer 2: ```suggestion
  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
```

* [#4983](https://github.com/osmosis-labs/osmosis/pull/4983) Consume a gas when creating a new token using tokenfactory as a spam deterrence mechanism.
* [#4951](https://github.com/osmosis-labs/osmosis/pull/4951) Implement pool liquidity query in pool manager, deprecate the one in gamm
* [#5138](https://github.com/osmosis-labs/osmosis/pull/5138) Rename swap fee to spread factor. Broke PoolI interface. Did not break CLI, message, pool models, queries or events.

> * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.

Reviewer 2: Is this actually an API break, or should it be under misc improvements instead?

  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook

>   * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0

Author: ```suggestion
```

  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook

>   * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0

Reviewer: This is duplicated under "Dependencies"

Reviewer: I think we should remove this

  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations

>   * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add

Reviewer: Also duplicated
```suggestion
```

>   * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add

Reviewer: this as well
```suggestion
```

  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add

>   * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.

Author: ```suggestion
  * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.
  * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod
```

  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook

>   * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0

Reviewer: duplicated
```suggestion
```

  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add
  * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.

>   * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod

Author: ```suggestion
```

### API breaks

> * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod

Reviewer: I don't see this change present

Author: we move it here: https://github.com/osmosis-labs/osmosis/pull/5418/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR98
